src/app/(app)/leaves/add/components/LeaveRequestFormBuilder.tsx

```diff
@@ -119,7 +110,13 @@ export const LeaveRequestFormBuilder = ({
     mode: "all",
   });
 
-  const { errors, isValid } = form.formState;
+  const {formState:{ errors, isValid } , setValue} = form;
+
+  useEffect(()=>{
+    if(loginUser?.role === MemberRole.Member){
+      setValue("memberId" , loginUser?.id.toString() ?? "")
```

Comment:
no need to this check `??` 
and i think you already handling this in separately